Fixes #2325: A11y + Highfi changes to audio player

rt4914 opened the pull request:
<!--
  - Thanks for submitting code to Oppia! Please fill out the following as part of
  - your pull request so we can review your code more easily.
  -->

## Explanation
<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->

Fixes #2325

Mocks: https://xd.adobe.com/view/dfe8ccd4-39eb-414d-a0b2-b5674357f966-8abb/screen/b73cb76c-3593-4a30-944a-ab240f00d0df/specs/

- As per the mocks the height of the audio player should be of 48dp. Now in previous implementation the height was more than 48dp because of incorrect calculations of height of the images. 
- This PR fixes the height of images by making changes to padding correctly. Also the bottom margin of root layout has been removed as it does not work across different layouts.
- Also the gap between the play/pause and seekbar start is also reduced.


## Note to reviewer
- To test this I suggest using `Galaxy Nexus` emulator as it almost resembles the mocks.
- If the below images look good to you than there is no need to check app for UI on device.
- This screen passes all A11y checks mentioned in #2325
- Talkback output has been added below.

## Before vs. After vs. Mock
<img width="250" alt="Before" src="https://user-images.githubusercontent.com/9396084/116427131-608bc980-a861-11eb-8327-0172298f908c.png">  <img width="250" alt="After" src="https://user-images.githubusercontent.com/9396084/116426900-24f0ff80-a861-11eb-8257-8f312a7f4751.png">  <img width="250" alt="Mock" src="https://user-images.githubusercontent.com/9396084/116426171-7fd62700-a860-11eb-8077-0e1214e7215d.png"> 

## Talkback output

https://user-images.githubusercontent.com/9396084/116435639-36d6a080-a869-11eb-8992-994cace28b71.mp4


## Espresso Test Result
<img width="450" alt="Screenshot 2021-04-30 at 10 14 20 AM" src="https://user-images.githubusercontent.com/9396084/116650474-cd58ae00-a99e-11eb-89af-71b9daf82e57.png">

Filed an issue for the above test failures because they are failing on develop too. Will try to to solve in separate PR.

<img width="450" alt="Screenshot 2021-04-30 at 10 33 35 AM" src="https://user-images.githubusercontent.com/9396084/116650781-8919dd80-a99f-11eb-8f6b-3500d2ac7c81.png">

There was one more test failing in `AudioFragmentTest` which is no longer needed therefore removed it in my last commit.

## Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

Commit:
Highfi changes to audio player

Commit:
Nit margin fix

Commit:
Accessiblity test clear + tests added

Commit:
Audio toolbar icon A11y checks

Commit:
Error fix

Commit:
Nit fix

Commit:
Fix test failure

Commit:
Removed test case

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/player/exploration/ExplorationActivityTest.kt:
```diff
@@ -209,6 +210,48 @@ class ExplorationActivityTest {
+  }
+
+  @Test
+  fun testExploration_clickAudioIcon_contentDescription_changesCorrectly() {
```
I suggest also adding a test for disabling again (to make sure it reverts back).

rt4914 replied:
Added test.

BenHenning reviewed (changes requested):
Thanks @rt4914! I think the solution looks good. Just have two additional questions besides my other comment from this review:
1. Can you post an indication whether the affected tests still pass on Espresso?
2. Is item 2 from #2325 not addressable/not planned to be addressed in this PR?

Commit:
Added test suggested by Ben

Commit:
Removed test

rt4914 commented:
> Thanks @rt4914! I think the solution looks good. Just have two additional questions besides my other comment from this review:
> 
> 1. Can you post an indication whether the affected tests still pass on Espresso?
> 2. Is item 2 from #2325 not addressable/not planned to be addressed in this PR?

1. I have added details in the description. Summary: Removed one test from AudioFragmentTest and filed #3145 as these tests are failing on develop.
2. It has been solved in #2519 by Arjun.

rt4914 commented:
> > Thanks @rt4914! I think the solution looks good. Just have two additional questions besides my other comment from this review:
> > 
> > 1. Can you post an indication whether the affected tests still pass on Espresso?
> > 2. Is item 2 from #2325 not addressable/not planned to be addressed in this PR?
> 
> 1. I have added details in the description. Summary: Removed one test from AudioFragmentTest and filed #3145 as these tests are failing on develop.
> 2. It has been solved in #2519 by Arjun.

@BenHenning The failing tests for `ExplorationActivityTest` have been fixed in https://github.com/oppia/oppia-android/pull/3147

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/player/exploration/ExplorationActivityTest.kt:
```diff
@@ -252,6 +252,30 @@ class ExplorationActivityTest {
+      explorationDataController.startPlayingExploration(FRACTIONS_EXPLORATION_ID_0)
+      networkConnectionUtil.setCurrentConnectionStatus(NetworkConnectionUtil.ConnectionStatus.LOCAL)
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.action_audio_player))
+        .perform(click())
+      onView(withId(R.id.action_audio_player))
+        .perform(click())
```
Prefer these to be on the same lines since line wrapping isn't needed here.

rt4914 replied:
Done.

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/player/exploration/ExplorationActivityTest.kt:
```diff
@@ -252,6 +252,30 @@ class ExplorationActivityTest {
 
+  @Test
+  fun testExploration_clickAudioIconTwice_contentDescription_changesToDefault() {
+    setupAudioForFraction()
```
Should be ``setUpAudioForFractionLesson`` ('setUp' being the main change). While not part of your PR, it'd be nice to fix this.

rt4914 replied:
Done.

BenHenning reviewed (approved):
Thanks @rt4914. Just 2 nits, otherwise PR LGTM.

Commit:
Nit fixes

Commit:
Merge remote-tracking branch 'upstream/develop' into highfi-audio-player

Commit:
Merge remote-tracking branch 'upstream/develop' into highfi-audio-player

rt4914 commented:
@BenHenning I am unable to understand why the bazel checks are failing.

BenHenning commented:
@rt4914 you probably already know this by now, but this is likely #3157.

Commit:
Merge remote-tracking branch 'upstream/develop' into highfi-audio-player

Commit:
Merge remote-tracking branch 'upstream/develop' into highfi-audio-player
